packages/playwright-core/src/server/isomorphic/utilityScriptSerializers.ts

```diff
@@ -68,6 +71,48 @@ export function source() {
     }
   }
 
+  function isTypedArray(obj: any, constructor: Function): boolean {
+    try {
+      return obj instanceof constructor || Object.prototype.toString.call(obj) === `[object ${constructor.name}]`;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  const typedArrayConstructors: Record<TypedArrayKind, Function> = {
+    i8: Int8Array,
+    ui8: Uint8Array,
+    ui8c: Uint8ClampedArray,
+    i16: Int16Array,
+    ui16: Uint16Array,
+    i32: Int32Array,
+    ui32: Uint32Array,
+    // TODO: add Float16Array once it's in baseline
+    f32: Float32Array,
+    f64: Float64Array,
+    bi64: BigInt64Array,
+    bui64: BigUint64Array,
+  };
+
+  function typedArrayToBase64(array: any) {
+    /**
+     * Firefox does not support iterating over typed arrays, so we use `.toBase64`.
```

Comment:
Perhaps instead of `Array.from()` use a for loop? This should be supported everywhere.

Reply:
Tried that, but doesn't work. It looks like any kind of iteration is forbidden:

https://searchfox.org/mozilla-central/source/js/xpconnect/wrappers/XrayWrapper.cpp#576

Reply:
But not inside the utility world, apparently 🤷